### PR TITLE
📖 book: update refs to older books

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
~ same as https://github.com/kubernetes-sigs/cluster-api/pull/6192 just for release-1.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Review by vincepri — APPROVED** [review]
/approve

File: docs/book/src/introduction.md [hunk 1]
```diff
@@ -11,8 +11,16 @@ Started by the Kubernetes Special Interest Group (SIG) [Cluster Lifecycle](https
 * [Developer guide](./developer/guide.md)
 * [Contributing](./CONTRIBUTING.md)
 
-**Using Cluster API v1alpha3?** See the [legacy
-documentation](https://release-0-3.cluster-api.sigs.k8s.io).
+<aside class="note">
+
+<h1>ClusterAPI documentation versions</h1>
+
+This book documents ClusterAPI v1.0. For other Cluster API versions please see the corresponding documentation:
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply docs/book/src/introduction.md:18]: Ideally at some point I want to have something like the version drop down on kubernetes.io: https://kubernetes.io/

But I think let's try to figure that out if we overhaul the whole page using e.g. hugo

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Pull request merged**